lix/packages/sdk/src/resolve-conflict/resolve-conflict.ts

```diff
@@ -0,0 +1,111 @@
+import type { Change, Conflict } from "../schema.js";
+import type { Lix } from "../types.js";
+import {
+	ChangeDoesNotBelongToFileError,
+	ChangeHasBeenMutatedError,
+	ChangeNotDirectChildOfConflictError,
+} from "./errors.js";
+
+/**
+ * Resolves a conflict by applying the given change.
+ */
+export async function resolveConflict(args: {
+	lix: Lix;
+	conflict: Conflict;
+	resolveWithChange: Change;
+}): Promise<void> {
+	if (args.lix.plugins.length !== 1) {
+		throw new Error("Unimplemented. Only one plugin is supported for now");
+	}
+
+	const plugin = args.lix.plugins[0];
+	if (plugin?.applyChanges === undefined) {
+		throw new Error(
+			"Plugin does not support applying changes and therefore cannot resolve conflicts",
+		);
+	}
+
+	const change = await args.lix.db
+		.selectFrom("change")
+		.selectAll()
+		.where("id", "=", args.conflict.change_id)
+		.executeTakeFirst();
+
+	const existingResolvedChange = await args.lix.db
+		.selectFrom("change")
+		.selectAll()
+		.where("id", "=", args.resolveWithChange.id)
+		.executeTakeFirst();
+
+	// verify that the existing change does not differ from the resolveWithChange
+	// (changes are immutable). A change that is the result of a merge resolution
+	// should be a new change.
+
+	if (
+		existingResolvedChange &&
+		JSON.stringify(existingResolvedChange) !==
```

Comment:
errror prone equality check.  deepequal would be better

Reply:
Will use deepequal. Thanks for flagging